### ISSUE TITLE
tf.keras.layers.Conv2DTranspose raises error when dilation rates are different in two dimensions

### ISSUE THREAD

**Issue by lugalUrim** [tensorflow/tensorflow#48469]
**System information**
- Have I written custom code (as opposed to using a stock example script provided in TensorFlow): Yes
- OS Platform and Distribution (e.g., Linux Ubuntu 16.04): Windows
- Mobile device (e.g. iPhone 8, Pixel 2, Samsung Galaxy) if the issue happens on mobile device: N/A
- TensorFlow installed from (source or binary): binary
- TensorFlow version (use command below): 2.4.1
- Python version: 3.7
- Bazel version (if compiling from source): N/A
- GCC/Compiler version (if compiling from source): N/A
- CUDA/cuDNN version: N/A
- GPU model and memory: N/A

**Describe the current behavior**
Conv2DTranspose raises AssertionError with diifferent dilation_rate for each dimension.

**Describe the expected behavior**
Expect no error.

**Standalone code to reproduce the issue**

```
import tensorflow as tf
x = tf.keras.Input([None, None, 16])
tf.keras.layers.Conv2DTranspose(filters=1, kernel_size=32, dilation_rate=(1,2))(x)
```

Output:

Traceback (most recent call last):
  File "...\venv\lib\site-packages\IPython\core\interactiveshell.py", line 3437, in run_code
    exec(code_obj, self.user_global_ns, self.user_ns)
  File "<ipython-input-32-469167c62d25>", line 1, in <module>
    tf.keras.layers.Conv2DTranspose(filters=1, kernel_size=32, dilation_rate=(1,2))(x)
  File "...\venv\lib\site-packages\tensorflow\python\keras\engine\base_layer.py", line 952, in __call__
    input_list)
  File "...\venv\lib\site-packages\tensorflow\python\keras\engine\base_layer.py", line 1091, in _functional_construction_call
    inputs, input_masks, args, kwargs)
  File "...\venv\lib\site-packages\tensorflow\python\keras\engine\base_layer.py", line 822, in _keras_tensor_symbolic_call
    return self._infer_output_signature(inputs, args, kwargs, input_masks)
  File "...\venv\lib\site-packages\tensorflow\python\keras\engine\base_layer.py", line 863, in _infer_output_signature
    outputs = call_fn(inputs, *args, **kwargs)
  File "...\venv\lib\site-packages\tensorflow\python\keras\layers\convolutional.py", line 1315, in call
    dilation_rate=self.dilation_rate)
  File "...\venv\lib\site-packages\tensorflow\python\util\dispatch.py", line 201, in wrapper
    return target(*args, **kwargs)
  File "...\venv\lib\site-packages\tensorflow\python\keras\backend.py", line 5355, in conv2d_transpose
    assert dilation_rate[0] == dilation_rate[1]
AssertionError

**Comment by UsharaniPagadala** [comment]
@lugalUrim 
As per the [documentation](https://www.tensorflow.org/api_docs/python/tf/keras/layers/Conv2DTranspose?version=nightly#expandable-1) stride and dilation_rate should be same,if stride=1,dilation_rate=(1,1) or if stride=2,dilation_rate=(2,2)
I [reproduced](https://colab.research.google.com/gist/UsharaniPagadala/37d997f5b9e66e26e75bacc5d323cdbe/untitled13.ipynb#scrollTo=mGOQcl_1pNK-) it by changing stride and dilation_rate as mentioned above. 
Could you please check and let us know if it helps.Thanks

**Comment by lugalUrim** [comment]
Thanks @UsharaniPagadala . I suppose you are suggesting that `dilation_rate` and `strides` should be the same for all spatial dimension? 

I was trying `dilation_rate=(1,2)` (without specifying strides) in order to have different dilation rate for width and height dimension. 

For example pytorch's Conv2d supports `dilation=(3, 1)`.
https://pytorch.org/docs/stable/generated/torch.nn.Conv2d.html#torch.nn.Conv2d

**Comment by lugalUrim** [comment]
I believe it's possible to have different dilation rates for width and height dimension. For example, in conv3d:
```
import tensorflow as tf
x = tf.keras.Input([None, None, None, 16])
tf.keras.layers.Conv3DTranspose(filters=1, kernel_size=32, dilation_rate=(1,2,1))(x)
```

**Comment by dengyinlin** [comment]
Hi everyone, any update on this issue?

**Comment by oawxkw** [comment]
Some alternative names of the API may also have the same issue, and users should be cautious when using them on both CPU and GPU up to tensorflow 2.8.4 (v2.8.3-90-g1b8f5c396f0):

> In tensorflow 2.6.1 (v2.6.0-101-g3aa40c3ce9d) and previous versions, it just throws an AssertionError Exception.
> In versions from tensorflow 2.7.0 (v2.7.0-rc1-69-gc256c071bb2) to 2.8.4 (v2.8.3-90-g1b8f5c396f0), it throws an Exception without any useful error message.

- `(tf.keras.layers.Conv2DTranspose)`, `tf.compat.v1.keras.layers.Conv2DTranspose`, `tf.compat.v1.keras.layers.Convolution2DTranspose`, `tf.keras.layers.Convolution2DTranspose`, `tf.compat.v1.layers.Conv2DTranspose`

<details>
  <summary>Code to reproduce the issue in <code>tf.compat.v1.keras.layers.Conv2DTranspose</code> APIs in older versions</summary>

```python
import tensorflow as tf
print(tf.version.GIT_VERSION, tf.version.VERSION, flush=True)
print(tf.config.list_physical_devices(), flush=True)


try:
    x = tf.keras.Input([None, None, 16])
    # tf.keras.layers.Conv2DTranspose(filters=1, kernel_size=32, dilation_rate=(1,2))(x)
    tf.compat.v1.keras.layers.Conv2DTranspose(filters=1, kernel_size=32, dilation_rate=(1,2))(x)
    # tf.compat.v1.keras.layers.Convolution2DTranspose(filters=1, kernel_size=32, dilation_rate=(1,2))(x)
    # tf.keras.layers.Convolution2DTranspose(filters=1, kernel_size=32, dilation_rate=(1,2))(x)
    # tf.compat.v1.layers.Conv2DTranspose(filters=1, kernel_size=32, dilation_rate=(1,2))(x)
except Exception as e:
    print("Failed! Error:", str(e), flush=True)
else:
    print("Success!", flush=True)
```

On my GPU machine, it throws an unclear Exception as follows:

```text
v2.8.3-90-g1b8f5c396f0 2.8.4
[PhysicalDevice(name='/physical_device:CPU:0', device_type='CPU'), PhysicalDevice(name='/physical_device:GPU:0', device_type='GPU')]
Failed! Error: Exception encountered when calling layer "conv2d_transpose" (type Conv2DTranspose).



Call arguments received:
  • inputs=tf.Tensor(shape=(None, None, None, 16), dtype=float32)
```

This behavior is also reproducible on my CPU machine:

```text
v2.8.3-90-g1b8f5c396f0 2.8.4
[PhysicalDevice(name='/physical_device:CPU:0', device_type='CPU')]
Failed! Error: Exception encountered when calling layer "conv2d_transpose" (type Conv2DTranspose).



Call arguments received:
  • inputs=tf.Tensor(shape=(None, None, None, 16), dtype=float32)
```
</details>

It seems to be fixed in tensorflow 2.9.0 (v2.9.0-rc2-42-g8a20d54a3c1) and later versions.
Now it throws an Exception as follows, and the error message is more clear:

```text
Expected the 2 dimensions of the `dilation_rate` argument to be equal to each other. Received: dilation_rate=(1, 2)
```

**Comment by tilakrayal** [comment]
@lugalUrim, @oawxkw,
I tried to execute the both mentioned codes with the latest stable v2.15 by importing the new keras version **keras_core** and the code was executed without any issues/error. Kindly find the gist of it [here](https://colab.research.google.com/gist/tilakrayal/8dcb920ab79032672f7e306addd6a153/untitled1564.ipynb). Thank you!
